client/components/tinymce/plugins/wpcom-view/views/embed/index.js

```diff
@@ -46,11 +41,11 @@ export default class EmbedViewManager extends EventEmitter {
 		}
 	}
 
-	addListener( event, listener ) {
+	addListener( event, listener, store ) {
 		super.addListener( event, listener );
-
 		if ( 'change' === event && 1 === this.listeners( event ).length ) {
-			sites.addListener( event, this.sitesListener );
+			this.store = store;
+			this.unsubscribeStore = store.subscribe( this.sitesListener );
```

Comment:
Hi @gwwar, I added functionality to the lib/middleware to subscribe to the selected site, and used it instead of subscribing to the store. It is a little hacky solution but it looks like its working ok, I would love to hear your feedback :)